tests/integration/duplicate_urns/step3/index.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Resource } from "./resource";
+
+// "a" is already in the snapshot and will be "same"d.
+const a = new Resource("a", { state: 4 });
+
+// "b" is not, but they have the same URN.
+const b = new Resource("a", { state: 5 });
```

Comment:
I don't exactly follow the comment here.  There is no "b" URN, and the "a" URN is in the snapshot.

Reply:
"b" is the name of the variable here and not the name of the Pulumi object.